src/coreclr/jit/importercalls.cpp

```diff
@@ -3364,6 +3366,28 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                                 retNode = gtNewFalse();
                             }
                             break;
+                        case NI_System_Type_get_IsGenericType:
+                        {
+                            // If the type is __Canon, we just give up on expanding the intrinsic
+                            if (info.compCompHnd->getClassAttribs(hClass) & CORINFO_FLG_SHAREDINST)
```

Comment:
```suggestion
                            if ((info.compCompHnd->getClassAttribs(hClass) & CORINFO_FLG_SHAREDINST) != 0)
```
(Preferred for new code)